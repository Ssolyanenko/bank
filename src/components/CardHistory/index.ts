export * from './History';
export * from './Pagination';
export * from './CardHistory';
