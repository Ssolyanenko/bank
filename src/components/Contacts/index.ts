export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
