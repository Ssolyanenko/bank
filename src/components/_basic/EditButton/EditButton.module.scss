@import 'src/styles/variables.module.scss';

.editBtn {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 12px;
  width: 48px;
  height: 48px;
  background: $orange;
  border-radius: 2px;
  border: 1px solid $orange;
  cursor: pointer;
  transition: 0.5s;

  .btn {
    fill: $grayDark_6;
  }

  &:hover {
    border-color: $brand_3;
    background-color: $brand_3;
  }

  &.small {
    width: 48px;
  }

  &.medium {
    width: 72px;
  }

  &.large {
    width: 92px;
  }
}
