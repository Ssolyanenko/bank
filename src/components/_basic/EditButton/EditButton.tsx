import React, { FC } from 'react';

import { EditIcon } from 'assets/EditIcon';
import { Size } from 'interfaces/common/componentsSettings';
import classes from './EditButton.module.scss';

interface Props {
  onClick?(): void;
  size?: Size;
  className?: string;
}

export const EditButton: FC<Props> = ({ onClick, size = Size.SMALL, className = '' }) => (
  <button className={`${classes.editBtn} ${className} ${classes[size]}`} type="button" onClick={onClick}>
    <EditIcon className={classes.btn} size={size} />
  </button>
);
