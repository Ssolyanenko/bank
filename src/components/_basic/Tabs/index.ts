export * from './Tabs';
export * from './TabTitle';
