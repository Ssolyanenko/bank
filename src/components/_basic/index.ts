export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
