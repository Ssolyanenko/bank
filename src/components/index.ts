export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
