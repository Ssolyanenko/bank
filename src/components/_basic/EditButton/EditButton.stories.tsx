import { ComponentMeta, ComponentStory } from '@storybook/react';

import { EditButton } from './EditButton';

export default {
  title: 'Basic/Buttons/EditButton',
  component: EditButton,
  parameters: {
    docs: {
      description: {
        component: 'Edit button for Notifications form',
      },
    },
  },
} as ComponentMeta<typeof EditButton>;

const Template: ComponentStory<typeof EditButton> = (props) => <EditButton {...props} />;

export const Default = Template.bind({});

Default.args = {
  onClick(): void {
    alert('click');
  },
};
