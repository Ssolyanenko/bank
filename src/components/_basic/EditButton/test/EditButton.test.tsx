import React from 'react';
import { shallow } from 'enzyme';

import { EditButton } from 'components/_basic';
import { Default } from 'components/_basic/EditButton/EditButton.stories';
import { Size } from 'interfaces/common/componentsSettings';

describe('<EditButton />', (): void => {
  it('should make a snapshot of component', (): void => {
    const props = {
      onClick: jest.fn(),
      size: Size.MEDIUM,
    };
    const wrapper = shallow(<EditButton {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should render EditButton from storybook', (): void => {
    const props = {
      onClick: jest.fn(),
      size: Size.MEDIUM,
    };

    const wrapper = shallow(<Default {...props} />);

    expect(wrapper.find(EditButton).length).toBe(1);
  });
});
