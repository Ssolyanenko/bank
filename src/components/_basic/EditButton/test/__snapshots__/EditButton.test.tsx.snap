// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditButton /> should make a snapshot of component 1`] = `
<button
  className="editBtn  medium"
  onClick={[MockFunction]}
  type="button"
>
  <EditIcon
    className="btn"
    size="medium"
  />
</button>
`;
